Core/ExchangeService.cs

```diff
@@ -2849,7 +2849,9 @@ public void UnpinTeamMailbox(EmailAddress emailAddress)
 
             EwsUtilities.ValidateParamCollection(folderIds, "folderIds");
 
-            return this.BuildSubscribeToStreamingNotificationsRequest(folderIds, eventTypes).Execute()[0].Subscription;
+            StreamingSubscription StreamingSub = this.BuildSubscribeToStreamingNotificationsRequest(folderIds, eventTypes).Execute()[0].Subscription;
+            StreamingSub.folderIds = folderIds;

```

Comment:
Looking at this a bit more, it seems that it might be more effective to expose the request object off of the StreamingSubscription.  This because I could see people being interested in the EventTypes as well.  So you would add a Request prop on StreamingSubscription and remove your folderIds one.